engines/director/sprite.cpp

```diff
@@ -455,19 +455,21 @@ void Sprite::setCast(CastMemberID memberID) {
 			}
 		}
 
-		Common::Rect dims = _cast->getInitialRect();
-		switch (_cast->_type) {
-		case kCastBitmap:
-			_width = dims.width();
-			_height = dims.height();
-			break;
-		case kCastShape:
-		case kCastText: 	// fall-through
-			break;
-		default:
-			_width = dims.width();
-			_height = dims.height();
-			break;
+		if (replaceDims) {
+			Common::Rect dims = _cast->getInitialRect();
+			switch (_cast->_type) {
+			case kCastBitmap:
```

Comment:
Would it make sense to combine `kCastBitmap:` with `default:`?

Reply:
Fixed